instance-scheduler/Dockerfile

```diff
@@ -1,16 +1,26 @@
+# Stage 1: Build the Go binary
 FROM public.ecr.aws/lambda/go:1 as builder
```

Comment:
Probably out of the scope of this work - but it looks like [lambda/go:1 is being deprecated](https://aws.amazon.com/cn/blogs/compute/migrating-aws-lambda-functions-from-the-go1-x-runtime-to-the-custom-runtime-on-amazon-linux-2/) in favor of `lambda/provided:al2` 🧓 🐳

Reply:
bug raised investigate further: https://github.com/ministryofjustice/modernisation-platform/issues/8027